component.go

```diff
@@ -292,6 +294,22 @@ type ListenerOptions struct {
 // correctness. Only use routing to increase performance in the common case.
 type WithRouter[T any] struct{}
 
+func (WithRouter[T]) routedBy(T) {}
+
+// RoutedBy[T] is the interface implemented by a struct that embeds
+// weaver.RoutedBy[T].
+type RoutedBy[T any] interface {
+	routedBy(T)
```

Comment:
This is really cool! :trophy: 

I thought Go didn't allow you to define multiple methods with the same name but different args/results. Looks like it does thanks to embeddings. And then they do this hide-y thing that most modern languages do. Very interesting!

Reply:
Thanks! Yeah, I also thought it was interesting how we can leverage Go's embedding semantics. Definitely was a brain teaser trying to figure this all out.